src/laboratory/signals.py

```diff
@@ -24,13 +25,10 @@ def shelf_object_base_quantity(sender, **kwargs):
     instance = kwargs.get('instance')
     if hasattr(instance, "measurement_unit") and hasattr(instance,"quantity"):
         try:
-            base_unit = BaseUnitValues.objects.get(
-                measurement_unit=instance.measurement_unit
-            )
-            quantity = Decimal(str(instance.quantity))
-            base_unit_value = Decimal(str(base_unit.si_value))
-            result = float(quantity / base_unit_value)
-            instance.quantity_base_unit = result
+
+                instance.quantity_base_unit = get_conversion_units(
+                    instance.measurement_unit, instance.quantity)
```

Comment:
You need to validate if this result is not None, instead of you need to replace for instance.quantity.